### ISSUE TITLE
salt state status.process always returns false

### ISSUE THREAD

**Issue by jak3kaj** [saltstack/salt#45188]
### Description of Issue/Question
The status.process salt state always returns false.  Testing with the salt module status.pid works as expected.

### Setup
```
admin@server:~$ cat /srv/salt/process-check.sls
Check if salt is running:
    status.process:
        - name: salt
```

### Steps to Reproduce Issue

This simple sls script just checks for the process name  "salt", which should match the salt-minion processes on the salt-minion; but the check always comes back false.

```
admin@server:~$ salt '*' state.sls process-check
test-minion:
----------
          ID: Check if ssh is running
    Function: status.process
        Name: salt
      Result: False
     Comment: Process signature "salt" not found
     Started: 01:47:35.468477
    Duration: 111.69 ms
     Changes:

Summary for test-minion
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time: 111.690 ms
ERROR: Minions returned with non-zero exit code
```

The module that status.process calls is status.pid, which returns the list of pids matching "salt".

```
admin@server:~$ salt '*' status.pid salt
test-minion:
    5272
    5278
    5281
    16892
    18658
```

### Versions Report
```
admin@server:~$ salt --versions-report
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 20 2017, 18:23:56)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-1044-aws
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by gtmanfred** [comment]
Hrm, there is definitely a bug there.

I have fixed this in #45199

Thanks,
Daniel

**Comment by jak3kaj** [comment]
Ok, thanks.  I also submitted  PR 45189 https://github.com/saltstack/salt/pull/45189

My PR is similar, but also doesn't return the data because I imagined this should return  True or False, and the pid list may be quite long for some checks.  What do you think about returning the PIDs?
